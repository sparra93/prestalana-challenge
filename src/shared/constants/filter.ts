import type { IFilterState } from '@/shared/Interfaces/Filter.interface';

export const filterSliceInitialState: IFilterState = {
  yearFilter: [],
  colorFilter: [],
  pantoneFilter: [],
  nameFilter: '',
};
