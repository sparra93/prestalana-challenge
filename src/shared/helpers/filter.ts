export const getUnique = <T>(values: T[], by: keyof T): string[] => {
  const uniques = values.map((value) => String(value[by]) as unknown as string);
  return Array.from(new Set(uniques)).sort();
};
