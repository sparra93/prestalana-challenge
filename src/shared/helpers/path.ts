export const ENDPOINT_URL = {
  LOGIN: {
    SIGIN: '/login',
    GETLIST: '/users?page=1&per_page=20',
    GETUSER: '/users/',
  },
  PRODUCT: {
    ALL: '/unknown?page=1&per_page=20',
  },
};

export function getUrl(path: string): string {
  return `${process.env.REMOTE_API_URL}${path}`;
}
