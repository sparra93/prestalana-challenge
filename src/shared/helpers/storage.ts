export function getFromLocalStorage(key: string) {
  return localStorage.getItem(key);
}
export function setLocalStorage(key: string, value: string) {
  return localStorage.setItem(key, value);
}
export function getFromSessionStorage(key: string) {
  return sessionStorage.getItem(key);
}
export function setSessionStorage(key: string, value: string) {
  return sessionStorage.setItem(key, value);
}
