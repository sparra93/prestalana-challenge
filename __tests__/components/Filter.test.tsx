import { fireEvent, screen } from '@testing-library/react';
import React from 'react';

import CheckBoxFilter from '@/components/Filter';
import { getUnique } from '@/shared/helpers/filter';

import { PRODUCT_MOCK } from '../../__mocks__/product.data';

describe('Filter Component', () => {
  const options = getUnique(PRODUCT_MOCK, 'year');
  const onChangeCheck = jest.fn();

  it('should display filter information correctly', () => {
    renderWithProviders(
      <CheckBoxFilter
        onChange={onChangeCheck}
        options={options}
        optionsSelected={[]}
      />,
    );

    const checkbox = screen.getByTestId(`checkbox-2001`);
    expect(checkbox).toBeInTheDocument();
  });

  it('should do click in the checkbox and call the function', () => {
    renderWithProviders(
      <CheckBoxFilter
        onChange={onChangeCheck}
        options={options}
        optionsSelected={[]}
      />,
    );

    const checkbox = screen.getByTestId(`checkbox-2001`);
    fireEvent.click(checkbox);
    expect(checkbox).toBeInTheDocument();
    expect(onChangeCheck).toHaveBeenCalledWith('2001');
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(
      <CheckBoxFilter
        onChange={onChangeCheck}
        options={options}
        optionsSelected={[]}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
