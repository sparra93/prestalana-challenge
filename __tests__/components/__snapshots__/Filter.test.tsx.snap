// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="sc-25792f0-0 gfAaqg"
  >
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2000"
        id="checkbox-2000"
        name="checkbox-2000"
        type="checkbox"
        value="2000"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2000"
      >
        2000
      </label>
    </li>
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2001"
        id="checkbox-2001"
        name="checkbox-2001"
        type="checkbox"
        value="2001"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2001"
      >
        2001
      </label>
    </li>
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2002"
        id="checkbox-2002"
        name="checkbox-2002"
        type="checkbox"
        value="2002"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2002"
      >
        2002
      </label>
    </li>
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2003"
        id="checkbox-2003"
        name="checkbox-2003"
        type="checkbox"
        value="2003"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2003"
      >
        2003
      </label>
    </li>
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2004"
        id="checkbox-2004"
        name="checkbox-2004"
        type="checkbox"
        value="2004"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2004"
      >
        2004
      </label>
    </li>
    <li
      class="sc-25792f0-1 krontY"
    >
      <input
        class="sc-25792f0-3 iUBIxP"
        data-testid="checkbox-2005"
        id="checkbox-2005"
        name="checkbox-2005"
        type="checkbox"
        value="2005"
      />
      <label
        class="sc-25792f0-2 jKYfwo"
        for="checkbox-2005"
      >
        2005
      </label>
    </li>
  </ul>
</DocumentFragment>
`;
